mp/src/game/client/momentum/ui/controls/DoubleLabel.cpp

```diff
@@ -0,0 +1,151 @@
+#include "cbase.h"
+
+#include "DoubleLabel.h"
+#include "util/mom_util.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+DoubleLabel::DoubleLabel(Panel *parent, const char *panelName) : BaseClass(parent, panelName, "")
+{
+    SetupSecondaryLabel(parent, panelName, "Secondary", "");
+}
+
+DoubleLabel::DoubleLabel(Panel *parent, const char *panelName, const char *secondPanelAppendedName) : BaseClass(parent, panelName, "")
+{
+    SetupSecondaryLabel(parent, panelName, secondPanelAppendedName, "");
+}
+
+DoubleLabel::DoubleLabel(Panel *parent, const char *panelName, const char *text1, const char *text2) : BaseClass(parent, panelName, text1)
+{
+    SetupSecondaryLabel(parent, panelName, "Secondary", text2);
+}
+
+DoubleLabel::DoubleLabel(Panel *parent, const char *panelName, const char *secondPanelAppendedName,
+                               const char *text1, const char *text2) : BaseClass(parent, panelName, text1)
+{
+    SetupSecondaryLabel(parent, panelName, secondPanelAppendedName, text2);
+}
+
+void DoubleLabel::SetupSecondaryLabel(Panel *parent, const char *panelName, const char *secondPanelAppendedName, const char *text)
+{
+    char secondaryPanelName[256];
+    Q_snprintf(secondaryPanelName, sizeof(secondaryPanelName), "%s%s", panelName, secondPanelAppendedName);
```

Comment:
I think this appending logic shouldn't be the responsibility of this class, but rather what constructs it. General controls like this should be pretty naive about what's using it.